flink-libraries/flink-table/src/main/java/org/apache/flink/api/table/functions/utils/MathFunctions.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.table.functions.utils;
+
+import java.math.BigDecimal;
+
+public class MathFunctions {
+
+	public static double power(double a, BigDecimal b) {
+		return Math.pow(a, b.doubleValue());
+	}
+
+}
```

Comment:
Why I didn't write `power` function in scala class.

I plunged into a problem with static access to method. I tried rewrite `MathFunctions.java` into Scala object https://goo.gl/rmLy8f . 
In BuiltInMethods we cannot to use `classOf[]` with scala `object`. So I used `getClass()`
Test throws [exception](http://pastebin.com/fLNr4CfX) 
because `getClass()` doesn't return `MathFunctions` itself, it returns `MathFunctions$` http://pastebin.com/LiZk45Fe 
**But** there is a solving :)
```
val POWER_DEC =
Types.lookupMethod(Class.forName(MathFunctions.getClass.getCanonicalName.dropRight(1)),
    "power", classOf[Double], classOf[JBigDecimal])
```
I prefer java approach, what do you think @wuchong @fhueske ?